src/js/components/IconButton.jsx

```diff
@@ -0,0 +1,82 @@
+import React, { PropTypes } from 'react';
+import { Icon } from '@jenkins-cd/react-material-icons';
+
+/**
+ * button element with icon.
+ * Use "iconName" for standard material-ui icons
+ * Use "children" for custom icon. Provide element (e.g. raw svg), or React component
+ *
+ * @param {object} [children] - React element or custom component to render as icon
+ * @param {string} [className] - custom class name for outer element
+ * @param {object} [style] - custom style object
+ * @param {string} [label] - button text
+ * @param {string} [iconName] - name of material-ui icon to display
+ * @param {number} [iconSize] - width/height of icon
+ * @param {string} [iconFill] - color code to apply as fill
+ * @param {function} [onClick] - onclick callback function
+ * @constructor
+ */
+export function IconButton({children, className, style, label, iconName, iconSize, iconFill, onClick}) {
```

Comment:
I'd really like to see some flow types for these sorts of config object params if nothing else.